src/hotspot/share/jvmci/jvmciCompilerToVM.cpp

```diff
@@ -1680,7 +1682,7 @@ C2V_VMENTRY_0(jint, isResolvedInvokeHandleInPool, (JNIEnv* env, jobject, ARGUMEN
       vmassert(!resolved_method->is_static(),"!");
       vmassert(MethodHandles::is_signature_polymorphic_method(resolved_method()),"!");
       vmassert(!MethodHandles::is_signature_polymorphic_static(resolved_method->intrinsic_id()), "!");
-      vmassert(cp_cache_entry->appendix_if_resolved(cp) == nullptr, "!");
+      vmassert(cp->cache()->appendix_if_resolved(index) == nullptr, "!");
```

Comment:
Another place where you have the `ResolvedMethodEntry` and are accessing via the cache with an index.